CMakeLists.txt

```
@@ -117,14 +117,26 @@ therock_add_feature(HIPIFY
 )
 
 # Core Features.
-therock_add_feature(CORE_RUNTIME
-  GROUP CORE
-  DESCRIPTION "Enables the core device runtime and tools"
-)
+# The HIP runtime for ROCm uses different implementations per platform today:
+#   * HIP on Linux uses the HSA runtime in ROCR-Runtime, used by CLR.
+#   * HIP on Windows uses the PAL runtime included directly as part of CLR.
+# https://rocm.docs.amd.com/projects/install-on-windows/en/latest/reference/component-support.html
+if(NOT WIN32)
+  # Other platforms (Linux) _do_ have the "core runtime".
+  therock_add_feature(CORE_RUNTIME
+    GROUP CORE
+    DESCRIPTION "Enables the core device runtime and tools"
+  )
+  set(_hip_runtime_platform_requirements "CORE_RUNTIME")
```

```suggestion
  list(APPEND _hip_runtime_platform_requirements "CORE_RUNTIME")
```
could work here if `set(_hip_runtime_platform_requirements` would be called before the if clause. That could allow to drop the else block.

---

The PR right now:

```cmake
if(NOT WIN32)
  set(_var "DEFAULT VALUES")
else()
  set(_var "WINDOWS VALUES")
endif()
```

Your suggestion:

```cmake
set(_var "")
if(NOT WIN32)
  list(APPEND _var "DEFAULT VALUES")
endif()
```

I like my explicit approach since it gives an obvious space for comments for both Linux and Windows, without needing to flip through some logic in your head.

WDYT?

---

Either works for me and I don't have strong feelings. Just suggesting as it might be more close to what we do for optional deps, see e.g.
https://github.com/ROCm/TheRock/blob/bc656e498e2dc3edce9aa5dac1e06fc363273681/math-libs/BLAS/CMakeLists.txt#L272-L275
even though this is more straight forward as it doesn't flips the logic with a `NOT`. Thus your explicit approach might be indeed better suited. Up to you :)